dbt/tests/data_tests/generic_tests/check_row_counts_per_partition.sql

```diff
@@ -1,32 +1,6 @@
 {% test check_row_counts_per_partition(model, table_name, partition_column) %}
 
-WITH
-    expected AS (
-        SELECT table_name, CAST(partition as VARCHAR) as partition, row_count as expected_count
-        {% if target.name == "etl-fast" %}
-        FROM {{ ref("etl_fast_row_counts") }}
-        {% else %}
-        FROM {{ ref("etl_full_row_counts") }}
-        {% endif %}
-        WHERE table_name = '{{ table_name }}'
-    ),
-    observed AS (
-        {% if partition_column == "report_year" %}
-        SELECT CAST({{ partition_column }} as VARCHAR) as partition, COUNT(*) as observed_count
-        FROM {{ model }}
-        GROUP BY {{ partition_column }}
-        {% elif partition_column in ["report_date", "datetime_utc"] %}
-        SELECT CAST(YEAR({{ partition_column }}) as VARCHAR) as partition, COUNT(*) as observed_count
-        FROM {{ model }}
-        GROUP BY YEAR({{ partition_column }})
-        {% else %}
-        SELECT '' as partition, COUNT(*) as observed_count
-        FROM {{ model }}
-        {% endif %}
-    )
-SELECT expected.partition, expected.expected_count, observed.observed_count
-FROM expected
-INNER JOIN observed ON expected.partition=observed.partition
-WHERE expected.expected_count != observed.observed_count
+-- note 2025-03-28: logic is in a macro so that we can test it.
+{{ row_counts_per_partition(model, table_name, partition_column) }}
```

Comment:
It would be nice to not need the test and macro files, but I don't really see a way around that. I could see this being a common pattern to test generic tests, though, so I'm trying to think if there's a better pattern to keep this all organized.

One thought that comes to mind would be to have a single big `generic_tests.sql` file and define all of the tests themselves as `macros` elsewhere, then have a `{{% test` block for each test that just calls the macros?

Reply:
Sort of like headers & source files in some other languages - I could see that working. Seems out of scope for this PR though :)